src/Tickets/Emails/Provider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * The main service provider for the Tickets Emails.
+ *
+ * @since   TBD
+ * @package TEC\Tickets\Emails
+ */
+
+namespace TEC\Tickets\Emails;
+
+use tad_DI52_ServiceProvider;
+
+/**
+ * Service provider for the Tickets Emails.
+ *
+ * @since   TBD
+ * @package TEC\Tickets\Emails
+ */
+class Provider extends tad_DI52_ServiceProvider {
+
+	/**
+	 * Register the provider singletons.
+	 *
+	 * @since TBD
+	 */
+	public function register() {
+
+		// If not enabled, do not load Tickets Emails system.
+		if ( ! tec_tickets_emails_is_enabled() ) {
+			return;
+		}
+
+		$this->register_assets();
+		$this->register_hooks();
+
+		// Register singletons.
+		$this->container->singleton( static::class, $this );
+		$this->container->singleton( 'tickets.emails.provider', $this );
+		$this->container->singleton( Emails_Tab::class );
+		$this->container->singleton( 'tickets.emails.emails_tab', $this );
```

Comment:
I think we're using hyphen for singletons. @bordoni can confirm.
```suggestion
		$this->container->singleton( 'tickets.emails.emails-tab', $this );
```

Reply:
Ah... I think you're right. I'll let @bordoni confirm, though, before I make the change.

Reply:
Looking at other instances in the code, I see where we are using hyphens, so I went ahead and pulled the trigger on the change. Thanks for bringing this up @juanfra!